CIP-0018/README.md

```diff
@@ -55,42 +56,42 @@ Key de-registrations work symmetrically and require that:
+- Carrying an extra UTxO pointer makes it possible to not worry (too much) about concurrency issues and problems coming with either, multiple instances of the wallet (like many users do between a mobile and desktop wallet) or the usual rollbacks which may otherwise create gaps in the indexes. By forcing all registration (resp. deregistration) transactions to be chained together, we also enforce that any rollbacks do maintain consistency of the index state: if any intermediate transaction is rolled back, then transactions they depend on are also rolled back. 
 
-We deem this to be an acceptable and fairly minor consequence but encourage existing software to raise awareness about this behaviour.
+- The first registration induces an extra cost for the end-user for the wallet needs to create a new UTxO with a minimum value. That UTxO is however passed from registration to registration afterwards without any extra cost. It can also be fully refunded upon de-registering the last stake key. So in practice, it works very much like a key deposit. 
 
-# Reference Implementation
+- We do not allow mixing up key registration and key deregistration as part of the same transaction for it makes the calculation of the pointer trickier for wallet processing transactions. A single transaction either move the pointer up or down. 
 
-Coming soon.
+- There's in principle nothing preventing someone from sending money to the special key-registration tracking address. Wallets should however only keep track of UTxOs created as part of transactions that register stake keys (and have therefore been authorized by the wallet itself). Applications are however encouraged to collect any funds sent to them in an ad-hoc manner on such keys. 
 
-# Glossary
+## Path to Active
 
-#### [1] Minimum Utxo Value
+### Acceptance Criteria
 
-The `minUTxOValue` is fixed by the protocol. It is defined as part of the Shelley genesis and can be updated via on-chain protocol updates. At the moment, this equals 1 Ada on the mainnet. 
+- [ ] There exists one or more reference implementations with appropriate testing illustrating the viability of this approach and specification.
 
-#### [2] Mangled Addresses
+### Implementation Plan
 
-An address is said mangled when it has a stake part, and the stake part isn't recognized as belonging to its associated wallet. That is, the payment part and the stake part appear to come from two different sources. This could be the case if the address has been purposely constructed in such a way (because the stake rights and funds are managed by separate entities), or because the stake part refers to a key hash which is no longer known of the wallet (because the associated key registration was rolled back).
+- [ ] Develop the proposed Reference Implementation as suggested when this CIP was originally published (see Discussion link in header for history).
```

Comment:
```suggestion
- [ ] Update this proposal to account for the Conway Ledger era, which brings new types of certificates for registering stake keys.

- [ ] Develop the proposed Reference Implementation as suggested when this CIP was originally published (see Discussion link in header for history).
```

Not sure if we want to add a note or something like this? to signal to readers that this CIP should be updated for the Conway ledger era. 
As Conway changes how stake keys can be registered, as well as adding new roles to stake keys (governance rights), which I believe impacts this CIP.

Reply:
yes @Ryun1 I think it's proactive & helpful to include that in the plan.